clients/client-python/tests/unittests/auth/test_oauth2_token_provider.py

```diff
@@ -0,0 +1,79 @@
+"""
+Copyright 2024 Datastrato Pvt Ltd.
+This software is licensed under the Apache License version 2.
+"""
+
+import unittest
+from unittest.mock import patch
+
+from gravitino.auth.auth_constants import AuthConstants
+from gravitino.auth.default_oauth_to_token_provider import DefaultOAuth2TokenProvider
+from tests.unittests.auth import mock_base
+
+OAUTH_PORT = 1082
+
+
+class TestOAuth2TokenProvider(unittest.TestCase):
```

Comment:
It's better to add an expired token test case, WDYT?

Reply:
Actually the last testcase is the test for expired token, but it does not so obvious.
https://github.com/noidname01/gravitino/blob/255cabdb882228f81de1a1907d34fba31cf5b351/clients/client-python/tests/unittests/auth/test_oauth2_token_provider.py#L65

The`old_access_token` will be loaded when `DefaultOAuth2TokenProvider` initializing, then the `old_access_token` will be replaced with `new_access_token` by calling `get_token_data()`

Reply:
I see. I think we also need to test abnormal cases, such as using an expired token to access the Server again, which should be rejected. I am not sure whether it is convenient to test the expired token to access the Server in a unit test. If it is not convenient, I think you can add a test case for authentication failure when using an expired token to access the Server in the integration tests.

Reply:
Yeah, it seems it's better to test in integration tests because it needs to interact with server.